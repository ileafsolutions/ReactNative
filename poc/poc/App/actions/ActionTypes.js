///Actions follow

export const ADD_TO_CART = "ADD_TO_CART"
export const FETCH_ITEMS = "FETCH_ITEMS"
export const LOADING = "LOADING"
