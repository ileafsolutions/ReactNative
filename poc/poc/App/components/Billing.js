import React from 'react';
import {
  View,
  Image,
  Text,
} from 'react-native';
import metrics from '../themes/Metrics'
import ListView from '../components/ListView'
import colors from '../themes/colors';
import images from '../images';

export default class Billing extends React.Component{

  render(){
    var responseData = this.props.cartList;
    console.log("responseData",responseData);
    var price=0;
    responseData.map( (item,index)=> {
      price = price + (item.price * item.qty);
      return item;
    });
    return(
      <View style ={{
        flex:1,
        flexDirection:'column',
        //paddingBottom:63,


      }}>
        <View style ={{flex:2,marginRight:10 }}>
          {this.renderifStatement(responseData.length)}
        </View>
        <View style ={{
          flex:1,
        //  justifyContent:'flex-end',
        }}>
          {this.renderBillRow('Sub Total', price)}
          {this.renderBillRow('Add Discount', '2%')}
          {this.renderBillRow('No Tax', '0.00')}
          {this.renderBillRow('Total', price)}
          {this.renderButton('PAY', '$ '+price)}
        </View>
      </View>
    )
  }

  renderifStatement(length){
    if(length==0){
      return(
        <View style = {{flex:1,padding:100,alignItems: 'center',justifyContent:'center'}}>
          <Text style = {{fontSize:16,textAlign:'center'}}>Search or use quick keys to add a product to your sale</Text>
        </View>
      )
    }else{
      return(
      <ListView>{this.props.cartList}</ListView>
    )
    }
  }
  renderBillRow(text,value){
    return(
      <View style ={{
        height:50,
        flexDirection:'row',
        alignItems:'center',
        backgroundColor: colors.colors_bill_bg,
        paddingLeft:15,
        paddingRight:15,
        borderBottomColor:colors.color_border_line,
        borderBottomWidth:2,
      }}>
        <Text style = {{
          flex:1,
          fontSize:20,
          color:text=='Add Discount' ? colors.color_discount_text : colors.color_bill_text,
        }}>{text}</Text>
        <Text style = {{
          flex:1,
          fontSize:20,
          color:text=='Add Discount' ? colors.color_discount_text : colors.color_bill_text,
          textAlign:'right'
        }}>{value}</Text>
      </View>
  )
  }

  renderButton(text,value){
    return(
      <View
        style ={{
          backgroundColor: colors.colors_button_bg,
          paddingLeft:10,
          paddingRight:10,
          height:85,
      }}>
        <Image
          style = {{
            flex:1,
            flexDirection:'row',
            width:(metrics.screenHeight/3)-20,
            resizeMode:'stretch',
            marginTop:15,
            marginBottom:15,
            alignItems:'center'
        }}
          source = {images.btn_bg}>

          <Text style = {{
            flex:0.65,
            fontSize:22,
            color:'white',
            borderRightColor:'#7CB938',
            borderRightWidth:1,
            marginLeft:10,
            backgroundColor:'transparent',
            fontWeight: 'bold',
          }}>{text}</Text>

          <Text style = {{
            flex:0.35,
            fontSize:22,
            color:'white',
            textAlign:'right',
            marginRight:20,
            backgroundColor:'transparent',
            fontWeight: 'bold',
          }}>{value}</Text>

        </Image>
      </View>
  )
  }
}
