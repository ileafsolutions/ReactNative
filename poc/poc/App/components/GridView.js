import React, { Component } from 'react'
import {
  AppRegistry,
  ListView,
  Image,
  StyleSheet,
  Text,
  TouchableOpacity,
  View
} from 'react-native'

import metrics from '../themes/Metrics'
const height = (metrics.screenWidth/3)
const width = metrics.screenHeight*2/3
const ds = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 })
export default class GridView extends Component {
  constructor(props) {
    super(props)
    const data = props.children || []

    this.state = {
      dataSource: ds.cloneWithRows(data)
    }

    this._renderRow = this._renderRow.bind(this)
    this._selectItem = this._selectItem.bind(this)
  }

  render() {
    return (
      <View style={styles.mainView}>
        <ListView
          contentContainerStyle={styles.list}
          dataSource={ds.cloneWithRows(this.props.children)}
          renderRow={this._renderRow}
          enableEmptySections={true}
          initialListSize={30}
        />
      </View>
    )
  }

  _renderRow (rowData, sectionId, rowId) {
    const imgSource = { uri: rowData.image }
    var name = rowData.base_name

    return (
      <TouchableOpacity
        style={styles.row}
        onPress={() => this._selectItem(rowData)}
        underlayColor='rgba(0,0,0,0)'>
        <View style={styles.rowComponent}>
          <Image style={styles.thumb} source={imgSource} />
          <Text style ={{
            flex:0.5,
            fontSize:18,
            textAlign:'center',
            paddingLeft:10,
            paddingRight:10,
            color:'black'
          }}>{name}</Text>
        </View>
      </TouchableOpacity>
    )
  }

  _selectItem (item) {
    // do something with item
    this.props.onItemPress(item)
  }
}

const styles = StyleSheet.create({
  mainView: {
    flex: 1,
    backgroundColor: 'white'
  },
  list: {
    backgroundColor:'white',
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  row: {
    width: (width/5)-30,
    height: width/4+30,
    margin:10
  },
  rowComponent: {
    width: (width/5)-30,
    height: width/4+30,
    alignItems: 'center'
  },
  thumb: {
    flex:0.75,
    width: (width/7),
    height: (width/7),
    resizeMode: 'contain'
  }
})
