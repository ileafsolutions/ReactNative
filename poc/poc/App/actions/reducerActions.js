import * as types from './ActionTypes'
var _ = require('lodash/core');

export function addToCart(item){
  //alert(JSON.stringify(item))
  return {
    type: types.ADD_TO_CART,
    item:item
  }
}
export function fetchItems(response){
  var res = response.products
  console.log("res",res);
  //_.values(response)
  return {
    type: types.FETCH_ITEMS,
    response:res
  }
}

export function apiToFetch(){
  return (dispatch,getState)=>{
  //  console.log("log for loadfing STARTED");
    dispatch(loading(true));
    //var c = new Date()
    //console.log("API CALL STARTED",c.getTime());
    fetch('https://projectpos.vendhq.com/api/products', {
            method: 'GET',
            headers: {
              'Accept': 'application/json',
              'Content-Type': 'application/json',
              'Authorization':'Bearer 5OtjwgBqfHJZh8NKxnSTP:fxR6cVw2aflNVf3U1l'
            }
          })
          .then((response) =>response.json())
          .then((responseJson) => {
            //var a = new Date()
          //  console.log("API parse to json ended",a.getTime());
            console.log("response",responseJson);
            console.log("log for loading Ended");
            dispatch(fetchItems(responseJson))
            dispatch(loading(false))
            ///below will be called atfter the api calls
            //dispatch(loadingForLogin())
            //dispatch(setToken(responseJson))
            //dispatch(navToMain())
          })
          .catch((error) => {
            console.log(error);
            dispatch(loading(false))
            //dispatch(setError())
          });
        }
}
export function loading(value){
  return {
    type: types.LOADING,
    value: value
  }
}
