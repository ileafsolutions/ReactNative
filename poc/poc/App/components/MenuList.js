import React from 'react';
import {
  View,
  Image,
  Text
} from 'react-native';
import GridView from './GridView';

//var response = require('../response.json');

import styles from '../containers/styles/RootContainerStyles'
import images from '../images';
const BASE_SIZE = 30;
const RANDOM_FACTOR_MAX = 30;

export default class MenuList extends React.Component{

  constructor(props) {
     super(props);
     const itemCount = 10;
     const data = []
     this.state = {
     data: []
    }
   }

  render(){

    var responseData = this.props.itemsList;
    console.log("responseNewData",responseData);
    /*for (let i = 0; i < responseData.length; i++) {
      this.state.data.push({
        id: i,
        uri: responseData.products[i].image,
        name: responseData.products[i].base_name,
        price:responseData.products[i].price,
        qty:1,
      })
    }*/
    var grid = responseData.map((item,index)=>{
      var newItem = item;
      return {
        ...item,
        qty:1
      }
    })
    console.log("GRID",grid);
    //console.log("daata",this.state.data);
    return(
      <View style ={{
        flex:1,
        paddingBottom:65,
        flexDirection:'column',
        marginLeft:10,
        backgroundColor:'white'
      }}>
        <GridView onItemPress={(item)=>{this.props.onItemPress(item)}}>{grid}</GridView>
      </View>
    )
  }

}
