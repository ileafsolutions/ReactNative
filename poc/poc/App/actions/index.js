import * as ReducerActions from './reducerActions'

export const ActionCreators = Object.assign({},ReducerActions,);
