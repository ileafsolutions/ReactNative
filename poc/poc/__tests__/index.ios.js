import 'react-native';
import React from 'react';
import Index from '../App/EntryPoint.js';
import GridView from '../App/components/GridView.js';
import MenuList from '../App/components/MenuList.js'
import Billing from '../App/components/Billing.js'
import * as actions from '../App/actions/reducerActions'
import * as types from '../App/actions/ActionTypes'
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Index />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
//
it('renders correctly', () => {
  const tree = renderer.create(
    <Billing cartList={[]}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
//MenuList need an input array named itemList to render correctly
it('renders correctly', () => {
  const tree = renderer.create(
    <MenuList itemsList={[]}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('actions', () => {
  it('should create an action to change the loader', () => {
    const value = true
    const expectedAction = {
      type: types.LOADING,
      value
    }
    expect(actions.loading(value)).toEqual(expectedAction)
  })
})


describe('actions', () => {
  it('should create an action to add an item to cart', () => {
    const item = {}
    const expectedAction = {
      type: types.ADD_TO_CART,
      item
    }
    expect(actions.addToCart(item)).toEqual(expectedAction)
  })
})

describe('actions', () => {
  it('should create an action to change the loader', () => {
    const res = {"products":[]}
    const response = res.products
    const expectedAction = {
      type: types.FETCH_ITEMS,
      response
    }
    expect(actions.fetchItems(res)).toEqual(expectedAction)
  })
})
