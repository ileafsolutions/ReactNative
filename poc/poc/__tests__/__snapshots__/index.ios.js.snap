exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <View
    style={
      Object {
        "flex": 1,
      }
    }>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
          },
          undefined,
        ]
      }>
      <View
        style={
          Object {
            "flex": 1,
          }
        }>
        <View
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "white",
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
              ],
            }
          }>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }>
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#2C353E",
                    "borderBottomColor": "#2C353E",
                    "flexDirection": "row",
                    "height": 64,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  }
                }>
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 32,
                      "marginRight": 15,
                      "resizeMode": "contain",
                      "width": 32,
                    }
                  } />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "white",
                      "flex": 1,
                      "fontSize": 19,
                    }
                  }>
                  MAIN OUTLET
                </Text>
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 22,
                      "resizeMode": "contain",
                      "width": 22,
                    }
                  } />
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 32,
                      "marginLeft": 15,
                      "resizeMode": "contain",
                      "width": 32,
                    }
                  } />
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#F5FCFF",
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }>
                <View
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderRightColor": "#EFEFEF",
                      "borderRightWidth": 2,
                      "height": 375,
                      "width": 444.6666666666667,
                    }
                  }>
                  <View
                    style={
                      Object {
                        "backgroundColor": "white",
                        "flex": 1,
                        "flexDirection": "column",
                        "marginLeft": 10,
                        "paddingBottom": 65,
                      }
                    }>
                    <View
                      style={
                        Object {
                          "backgroundColor": "white",
                          "flex": 1,
                        }
                      }>
                      <ScrollView
                        contentContainerStyle={
                          Object {
                            "backgroundColor": "white",
                            "flexDirection": "row",
                            "flexWrap": "wrap",
                          }
                        }
                        dataSource={
                          ListViewDataSource {
                            "items": 0,
                          }
                        }
                        enableEmptySections={true}
                        initialListSize={30}
                        onContentSizeChange={[Function]}
                        onEndReachedThreshold={1000}
                        onKeyboardDidHide={undefined}
                        onKeyboardDidShow={undefined}
                        onKeyboardWillHide={undefined}
                        onKeyboardWillShow={undefined}
                        onLayout={[Function]}
                        onScroll={[Function]}
                        pageSize={1}
                        removeClippedSubviews={true}
                        renderRow={[Function]}
                        scrollEventThrottle={50}
                        scrollRenderAheadDistance={1000}
                        stickyHeaderIndices={Array []} />
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "backgroundColor": "white",
                      "height": 245,
                      "width": 222.33333333333334,
                    }
                  }>
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column",
                      }
                    }>
                    <View
                      style={
                        Object {
                          "flex": 2,
                          "marginRight": 10,
                        }
                      }>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "justifyContent": "center",
                            "padding": 100,
                          }
                        }>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "fontSize": 16,
                              "textAlign": "center",
                            }
                          }>
                          Search or use quick keys to add a product to your sale
                        </Text>
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#2C353E",
                            "borderBottomColor": "#36414E",
                            "borderBottomWidth": 2,
                            "flexDirection": "row",
                            "height": 50,
                            "paddingLeft": 15,
                            "paddingRight": 15,
                          }
                        }>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "color": "#9DABBA",
                              "flex": 1,
                              "fontSize": 20,
                            }
                          }>
                          Sub Total
                        </Text>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "color": "#9DABBA",
                              "flex": 1,
                              "fontSize": 20,
                              "textAlign": "right",
                            }
                          }>
                          0
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#2C353E",
                            "borderBottomColor": "#36414E",
                            "borderBottomWidth": 2,
                            "flexDirection": "row",
                            "height": 50,
                            "paddingLeft": 15,
                            "paddingRight": 15,
                          }
                        }>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "color": "#4D7F88",
                              "flex": 1,
                              "fontSize": 20,
                            }
                          }>
                          Add Discount
                        </Text>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "color": "#4D7F88",
                              "flex": 1,
                              "fontSize": 20,
                              "textAlign": "right",
                            }
                          }>
                          2%
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#2C353E",
                            "borderBottomColor": "#36414E",
                            "borderBottomWidth": 2,
                            "flexDirection": "row",
                            "height": 50,
                            "paddingLeft": 15,
                            "paddingRight": 15,
                          }
                        }>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "color": "#9DABBA",
                              "flex": 1,
                              "fontSize": 20,
                            }
                          }>
                          No Tax
                        </Text>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "color": "#9DABBA",
                              "flex": 1,
                              "fontSize": 20,
                              "textAlign": "right",
                            }
                          }>
                          0.00
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#2C353E",
                            "borderBottomColor": "#36414E",
                            "borderBottomWidth": 2,
                            "flexDirection": "row",
                            "height": 50,
                            "paddingLeft": 15,
                            "paddingRight": 15,
                          }
                        }>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "color": "#9DABBA",
                              "flex": 1,
                              "fontSize": 20,
                            }
                          }>
                          Total
                        </Text>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "color": "#9DABBA",
                              "flex": 1,
                              "fontSize": 20,
                              "textAlign": "right",
                            }
                          }>
                          0
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "backgroundColor": "#232732",
                            "height": 85,
                            "paddingLeft": 10,
                            "paddingRight": 10,
                          }
                        }>
                        <Image
                          source={1}
                          style={
                            Object {
                              "alignItems": "center",
                              "flex": 1,
                              "flexDirection": "row",
                              "marginBottom": 15,
                              "marginTop": 15,
                              "resizeMode": "stretch",
                              "width": 202.33333333333334,
                            }
                          }>
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "borderRightColor": "#7CB938",
                                "borderRightWidth": 1,
                                "color": "white",
                                "flex": 0.65,
                                "fontSize": 22,
                                "fontWeight": "bold",
                                "marginLeft": 10,
                              }
                            }>
                            PAY
                          </Text>
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "color": "white",
                                "flex": 0.35,
                                "fontSize": 22,
                                "fontWeight": "bold",
                                "marginRight": 20,
                                "textAlign": "right",
                              }
                            }>
                            $ 0
                          </Text>
                        </Image>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`test renders correctly 2`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }>
  <View
    style={
      Object {
        "flex": 2,
        "marginRight": 10,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "padding": 100,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "textAlign": "center",
          }
        }>
        Search or use quick keys to add a product to your sale
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2C353E",
          "borderBottomColor": "#36414E",
          "borderBottomWidth": 2,
          "flexDirection": "row",
          "height": 50,
          "paddingLeft": 15,
          "paddingRight": 15,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#9DABBA",
            "flex": 1,
            "fontSize": 20,
          }
        }>
        Sub Total
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#9DABBA",
            "flex": 1,
            "fontSize": 20,
            "textAlign": "right",
          }
        }>
        0
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2C353E",
          "borderBottomColor": "#36414E",
          "borderBottomWidth": 2,
          "flexDirection": "row",
          "height": 50,
          "paddingLeft": 15,
          "paddingRight": 15,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#4D7F88",
            "flex": 1,
            "fontSize": 20,
          }
        }>
        Add Discount
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#4D7F88",
            "flex": 1,
            "fontSize": 20,
            "textAlign": "right",
          }
        }>
        2%
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2C353E",
          "borderBottomColor": "#36414E",
          "borderBottomWidth": 2,
          "flexDirection": "row",
          "height": 50,
          "paddingLeft": 15,
          "paddingRight": 15,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#9DABBA",
            "flex": 1,
            "fontSize": 20,
          }
        }>
        No Tax
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#9DABBA",
            "flex": 1,
            "fontSize": 20,
            "textAlign": "right",
          }
        }>
        0.00
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2C353E",
          "borderBottomColor": "#36414E",
          "borderBottomWidth": 2,
          "flexDirection": "row",
          "height": 50,
          "paddingLeft": 15,
          "paddingRight": 15,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#9DABBA",
            "flex": 1,
            "fontSize": 20,
          }
        }>
        Total
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#9DABBA",
            "flex": 1,
            "fontSize": 20,
            "textAlign": "right",
          }
        }>
        0
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#232732",
          "height": 85,
          "paddingLeft": 10,
          "paddingRight": 10,
        }
      }>
      <Image
        source={1}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "marginBottom": 15,
            "marginTop": 15,
            "resizeMode": "stretch",
            "width": 202.33333333333334,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRightColor": "#7CB938",
              "borderRightWidth": 1,
              "color": "white",
              "flex": 0.65,
              "fontSize": 22,
              "fontWeight": "bold",
              "marginLeft": 10,
            }
          }>
          PAY
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "flex": 0.35,
              "fontSize": 22,
              "fontWeight": "bold",
              "marginRight": 20,
              "textAlign": "right",
            }
          }>
          $ 0
        </Text>
      </Image>
    </View>
  </View>
</View>
`;

exports[`test renders correctly 3`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
      "marginLeft": 10,
      "paddingBottom": 65,
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
      }
    }>
    <ScrollView
      contentContainerStyle={
        Object {
          "backgroundColor": "white",
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
      dataSource={
        ListViewDataSource {
          "items": 0,
        }
      }
      enableEmptySections={true}
      initialListSize={30}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={1000}
      onKeyboardDidHide={undefined}
      onKeyboardDidShow={undefined}
      onKeyboardWillHide={undefined}
      onKeyboardWillShow={undefined}
      onLayout={[Function]}
      onScroll={[Function]}
      pageSize={1}
      removeClippedSubviews={true}
      renderRow={[Function]}
      scrollEventThrottle={50}
      scrollRenderAheadDistance={1000}
      stickyHeaderIndices={Array []} />
  </View>
</View>
`;
